insights/parsers/libssh_config.py

```diff
@@ -0,0 +1,128 @@
+"""
+Libssh configuration files
+==========================
+
+This module provides the ssh options in libssh configuration file
+
+Parsers provided by this module are:
+
+LibsshClientConfig - file ``/etc/libssh/libssh_client.config``
+--------------------------------------------------------------
+
+LibsshServerConfig - file ``/etc/libssh/libssh_server.config``
+--------------------------------------------------------------
+"""
+
+from insights.specs import Specs
+from insights.core import CommandParser
+from .. import parser, get_active_lines
+from insights.parsers import SkipException
+
+
+class LibsshConfig(CommandParser):
+    """
+    Base class for libssh configuration file.
+
+    There are many options in the libssh configuration file and
+    all of them are stored in one of the following formations:
+        a)   Key  [space]  Value
+        b)   Key  =  Value
+
+    There are the keywords used in the configuration file:
+        include
+        hostkey
+        listenaddress
+        port
+        loglevel
+        ciphers
+        macs
+        kexalgorithms
+        match
+        pubkeyacceptedkeytypes
+        hostkeyalgorithms
+        all
+        user
+        group
+        host
+        localaddress
+        localport
+        rdomain
+        address
+
+    Sample output::
+
+        # Parse system-wide crypto configuration file
+        Include /etc/crypto-policies/back-ends/libssh.config
+        # Parse OpenSSH configuration file for consistency
+        Include /etc/ssh/sshd_config
+
+    Attributes:
+        data (dict): store each line into a dict which has the key from
+        the first field of the line and the value from the second field
+
+    .. note::
+        If there are two or more lines that have the same key, then we
+        store the values in a list.
+
+    Examples:
+        >>> 'Include' in config.data
+        True
+        >>> config.data['Include']
+        ['/etc/crypto-policies/back-ends/libssh.config', '/etc/ssh/sshd_config']
+
+    Raises:
+        SkipException: When input content is empty or there is a syntax error.
+    """
+
+    def parse_content(self, content):
+        kv_pairs = {}
+
+        if not content:
+            raise SkipException('No content.')
+
+        for line in get_active_lines(content):
+            delimiter = None
+            if ' ' in line:
+                delimiter = ' '
+
+            if '=' in line:
+                delimiter = '='
+
+            try:
+                k, v = [i.strip() for i in line.split(delimiter)]
+                if k not in kv_pairs:
+                    kv_pairs[k] = v
+                else:
+                    _v = kv_pairs[k]
+                    _v = [_v] if not isinstance(_v, list) else _v
+                    if v not in _v:
+                        _v.append(v)
+                        kv_pairs[k] = _v
+            except ValueError:
+                raise SkipException('Syntax error')
+
+        self.data = kv_pairs
```

Comment:
Since the only attribute you are providing in this class is `data` which is a `dict`, it would be better to implement this class as a `dict`.  An example of how this works is provided in the [DnfModuleList](https://github.com/RedHatInsights/insights-core/blob/7809af44df433a2a70e42af2b2a724de072719cf/insights/parsers/dnf_module.py#L64) parser and several others.